### PR TITLE
Add search osc option

### PR DESCRIPTION
**Pull request by thgrund**
I added a new osc option to the plugin. 

With the `search` parameter, you are able to to move the cursort to the first occurance that matches the search term.

Fun fact: I forked the `tidalcycles/pulsar-tidalcycles` repository and Github changed it to `thgrund/atom-tidalcycles`. xD

### CODE REVIEW

**Comment by thgrund** [comment]
If I didn't made a mistake, then I solved the conflict in my fork and here, but github doesn't recognized it at solved and I do not have the write access to this repository to solve this.

**Comment by ndr-brt** [comment]
I think the issue here is that you pushed on the `master` branch of your fork and compared that one with upsteam's one.
please do your changes on a branch on your fork and then open the pr from that branch to the master of upstream

File: README.md [hunk 1]
```diff
@@ -105,6 +105,9 @@ Mandatory `type` argument to specify what kind of evaluation is requested:
 ##### Row / Column
 `row` and `column` parameters can be specified to move the cursor on that position before the evaluation.
 
+#### Search
+With the `search` parameter, you are able to to move the cursort to the first occurance that matches the search term.
```
Comment (ndr-brt) [inline README.md:109]:
```suggestion
With the `search` parameter, you are able to to move the cursor to the first occurrence that matches the search term.
```

File: lib/osc-loader.js [hunk 2]
```diff
@@ -30,6 +30,18 @@ export default class OscLoader extends EventEmitter {
...
+                }
+
+                if (resultDict['search']) {
+                    const lineNumber = this.editors.currentEditor().getText().split("\n").findIndex(line => line.includes(resultDict['search']));
+                    if (lineNumber > -1)  {
+                      this.editors.goTo(lineNumber, 0)
+                      this.editors.currentEditor().setScrollTopRow(lineNumber)
+                    }
+                }
+
```
Comment (ndr-brt) [inline lib/osc-loader.js:44]: I know that osc-loader has never been tested, but it would be possible to add a test for this feature?

**Comment by thgrund** [comment]
> I think the issue here is that you pushed on the `master` branch of your fork and compared that one with upsteam's one. please do your changes on a branch on your fork and then open the pr from that branch against the master of upstream

Oh gosh...if this is the error then I'm a bit embarrassed. Will fix this asap

File: lib/osc-loader.js (hunk 2 above)
Comment (thgrund) [reply lib/osc-loader.js:44]: I will have a look. Shouldn't be a big deal

**Comment by ndr-brt** [comment]
hi @thgrund , any chance to get this fixed?

**Comment by thgrund** [comment]
I apologize for being so quiet here. I didn't expect to have to leave this topic for so long. 
I'm currently customizing my forked version of the TidalCycles package and working heavily with sending and receiving OSC messages. What we have in the official repo is a basic requirement. So I also have a motivation that the whole thing is tested and works as expected.

**Comment by thgrund** [comment]
Hey @ndr-brt I will close this PR for now. My local development proceeded and I need time to work on the sharable "official" part of this. Sorry that it takes so long on my end...

**Pull request closed without merging**